### PR TITLE
CI: use scons cache mount for building simulator and docs

### PR DESCRIPTION
**Pull request by jnewb1**
depends:
https://github.com/commaai/openpilot/pull/29434
https://github.com/commaai/openpilot/pull/29437

### CODE REVIEW

File: tools/ci/Dockerfile.scons_cache
[the remark this answers is not in the document]
Comment (adeebshihadeh) [reply tools/ci/Dockerfile.scons_cache:1]: they should go in selfdrive/test/, which will be moved to tests/ pretty soon

[the remark this answers is not in the document]
Comment (jnewb1) [reply tools/ci/Dockerfile.scons_cache:1]: done

**Comment by jnewb1** [comment]
@adeebshihadeh this brings docs and sim builds to under 10 mins

**Comment by jnewb1** [comment]
Wait actually it looks like the moc files aren't cached every time so it's more like 11 sometimes, will look into that

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
docs is 12m and simulator is 11m. there is fundamentally no reason these are much slower than the `build all` job that took 5m. the goal is to make these as fast as we reasonably can

**Comment by jnewb1** [comment]
sim is down to 8:30 and docs is 7:30 now

**Comment by jnewb1** [comment]
Body and Panda aren't cached because git rev-parse fails in the container, since we don't copy the full git repo over. I'll handle that in a separate PR.

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**